torch_geometric/transforms/add_node_feature.py

```diff
@@ -0,0 +1,56 @@
+from typing import Tuple, Union
+
+import torch
+from torch import Tensor
+
+from torch_geometric.data import Data, HeteroData
+from torch_geometric.data.datapipes import functional_transform
+from torch_geometric.transforms import BaseTransform
+
+
+@functional_transform('add_node_feature')
+class AddNodeFeature(BaseTransform):
+    r"""Adds the node feature to :obj:`attr`
+    (functional name: :obj:`add_node_feature`).
+
+    Args:
+        value (str, Tensor, optional): The value of the attribute to add.
```

Comment:
What's the use-case of `value == Tensor`? Since transforms should be data-agnostic (i.e., independent from the number of nodes), how would this be used?

Reply:
`value == Tensor` is to refactor `AddLaplacianEigenvectorPE` and `AddRandomWalkPE` and address the TODO in `add_node_attr`. I'm also not sure this is indeed necessary.

Reply:
I think the idea would be more to have this as a utility function in `BaseTransform` so that every transform can make use of it. WDYT?

Reply:
Agreed.